### PR TITLE
feat: add documentation page support for V4 APIs

### PR DESCRIPTION
**Pull request by kamiiiel**
see https://gravitee.atlassian.net/browse/GKO-32
And https://gravitee.atlassian.net/browse/GKO-240

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-zqqajuudjf.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/model/crd/PageCRD.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    // Folder attributes
+    @With
+    private Boolean hidden;
+
+    // Non-Folder attributes
+    private String content;
+    private boolean homepage;
+
+    // Legacy support
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/model/crd/PageCRD.java]: I think we can remove this comment

**Review by a-cordier — COMMENTED** [review]
From fist test I did, it looks like the HTTP fetcher CRD is not working with V4 (content is not fetched

**Comment by kamiiiel** [comment]
> From fist test I did, it looks like the HTTP fetcher CRD is not working with V4 (content is not fetched

It is not supported at the moment actually. In the UI it is just saying that it will be available soon

**Comment by a-cordier** [comment]
> > From fist test I did, it looks like the HTTP fetcher CRD is not working with V4 (content is not fetched
> 
> It is not supported at the moment actually. In the UI it is just saying that it will be available soon

should be good with https://github.com/gravitee-io/gravitee-api-management/pull/7723/files

but then we need to add 
```java
pageSourceDomainService.setContentFromSource(page);
```

[here](https://github.com/gravitee-io/gravitee-api-management/pull/7648/files#diff-7d7687cebf2947972fdbb67eec7d6dd3f61a96f509956847613de4e95657d0e0R113)

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/domain_service/DocumentationValidationDomainService.java [hunk 2]
```diff
@@ -116,6 +110,26 @@ public Page validateAndSanitizeForCreation(Page page, String organizationId, boo
         return sanitizedPage;
     }
 
+    public Page validateAndSanitizeForUpdate(Page page, String organizationId, boolean shouldValidateParentId) {
+        validatePageContent(organizationId, page);
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/domain_service/DocumentationValidationDomainService.java:118]: can we add this satement before calling validatePageContent ?

```java
pageSourceDomainService.setContentFromSource(page);
```

**Review by a-cordier — CHANGES_REQUESTED** [review]

**Review by a-cordier — APPROVED** [review]

**Pull request merged**